src/main/business/applicationFacade.js

```diff
@@ -167,10 +170,20 @@ class ApplicationFacade{
      *                 - if the user named username is not a lecturerUsername (is not assigned to a course)
      *                 - if there is no user named TAUsername
      */
-    addTA(data){
-        this.userController.verifyLecturer(data.callingUser.username);
-        this.userController.verifyUserRegistered(data.username)
-        this.taskController.newTARequestTask(data.username);
+    async addTA(data){
+        await this.userController.updateStaff({
+            callingUser: data.callingUser,
+            username: data.username,
+            userType: USER_TYPES.TA,
+        })
+    }
+
+    async addLecturer(data){
+        await this.userController.updateStaff({
```

Comment:
I think this is not the best way, you may miss keys of the "data" object.

You should use 
{...data, otherFields}

Reply:
נעשה בכוונה, המטרה היא רק לעדכן את המשתמש להיות מרצה. אני מעוניין להוריד שינויים נוספים אם נדחפו